src/rules/groupedImportsRule.ts

```diff
@@ -0,0 +1,152 @@
+    private lastImportStatement: ImportStatement;
+
+    private static getImportStatementType(statement: ts.Statement): ImportStatementType {
+        const path = Walker.getImportPath(statement);
+        if (path.charAt(0) === ".") {
+            if (path.charAt(1) === ".") {
+                return ImportStatementType.PARENT_DIRECTORY_IMPORT;
+            } else {
+                return ImportStatementType.CURRENT_DIRECTORY_IMPORT;
+            }
+        } else {
+            return ImportStatementType.LIBRARY_IMPORT;
+        }
+    }
+
+    private static getImportPath(statement: ts.Statement): string {
+        const str = statement.getText();
+        let index;
+        let lastIndex;
+        index = str.indexOf("'");
+        if (index > 0) {
+            lastIndex = str.lastIndexOf("'");
+        } else {
+            index = str.indexOf("\"");
+            lastIndex = str.lastIndexOf("\"");
+        }
+        if (index < 0 || lastIndex < 0) {
+            throw new Error(`Unable to extract path from import statement \`${statement.getText()}\``);
+        }
+        return str.substring(index + 1, lastIndex);
+    }
+
+    public walk(sourceFile: ts.SourceFile): void {
+        sourceFile.statements
+            .filter(isImportDeclaration)
+            .forEach((st) => this.checkStatement(st));
+    }
+
+    private toImportStatement(statement: ts.Statement): ImportStatement {
+        return {
+            lineEnd: this.sourceFile.getLineAndCharacterOfPosition(statement.getEnd()).line,
+            lineStart: this.sourceFile.getLineAndCharacterOfPosition(statement.getStart()).line,
+            statement,
+            type: Walker.getImportStatementType(statement),
+        };
+    }
+
+    private checkStatement(statement: ts.Statement): void {
+        const importStatement = this.toImportStatement(statement);
+        if (this.lastImportStatement) {
+            this.checkImportStatement(importStatement);
+        }
+        this.lastImportStatement = importStatement;
+    }
+
+    private checkImportStatement(importStatement: ImportStatement) {
+        if (importStatement.type === this.lastImportStatement.type) {
+            if (importStatement.lineStart !== this.lastImportStatement.lineEnd + 1) {
+                const replacement = Lint.Replacement.deleteFromTo(
+                    this.lastImportStatement.statement.getEnd() + 1, importStatement.statement.getStart());
+                this.addFailureAtNode(importStatement.statement, Rule.IMPORT_SOURCES_SEPARATED, replacement);
+            }
+        } else if (importStatement.type.valueOf() < this.lastImportStatement.type.valueOf()) {
+            this.addFailureAtNode(importStatement.statement, Rule.IMPORT_SOURCES_ORDER, this.getAllImportsFix());
+        } else {
+            if (importStatement.lineStart !== this.lastImportStatement.lineEnd + 2) {
+                const replacement = Lint.Replacement.appendText(importStatement.statement.getStart(), ts.sys.newLine);
```

Comment:
`ts.sys.newLine` does not always match the line endings in the current file.
Have a look at `eoflineRule.ts` to see how the line ending is inferred there.